src/battle/overlay_12_0224E4FC.c

```diff
@@ -5541,3 +5542,842 @@ BOOL TryFling(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId) {
+                        break;
+            }
+        }
+        if (ctx->battleMons[ctx->battlerIdTemp].species == SPECIES_CHERRIM && ctx->battleMons[ctx->battlerIdTemp].hp) {
+            if (!CheckAbilityActive(bsys, ctx,CHECK_ABILITY_ALL_HP, 0, ABILITY_CLOUD_NINE) && !CheckAbilityActive(bsys, ctx, CHECK_ABILITY_ALL_HP, 0, ABILITY_AIR_LOCK)) {
+                if (!(ctx->fieldCondition & FIELD_CONDITION_WEATHER_CASTFORM) && ctx->battleMons[ctx->battlerIdTemp].form == (u8) CHERRIM_SUNNY) {
+                    ctx->battleMons[ctx->battlerIdTemp].form = (u8) CHERRIM_CLOUDY;
+                    *script = 262;
+                    ret = TRUE;
+                    break;
+                } else if ((ctx->fieldCondition & FIELD_CONDITION_SUN_ALL) && ctx->battleMons[ctx->battlerIdTemp].form == (u8) CHERRIM_CLOUDY) {
+                    ctx->battleMons[ctx->battlerIdTemp].form = (u8) CHERRIM_SUNNY;
+                    *script = 262;
+                    ret = TRUE;
+                    break;
+                } else if ((ctx->fieldCondition & FIELD_CONDITION_RAIN_ALL) && ctx->battleMons[ctx->battlerIdTemp].form == (u8) CHERRIM_SUNNY) {
+                    ctx->battleMons[ctx->battlerIdTemp].form = (u8) CHERRIM_CLOUDY;
+                    *script = 262;
+                    ret = TRUE;
+                    break;
+                } else if ((ctx->fieldCondition & FIELD_CONDITION_HAIL_ALL) && ctx->battleMons[ctx->battlerIdTemp].form == (u8) CHERRIM_SUNNY) {
+                    ctx->battleMons[ctx->battlerIdTemp].form = (u8) CHERRIM_CLOUDY;
+                    *script = 262;
+                    ret = TRUE;
+                    break;
+                }
+            } else if (ctx->battleMons[ctx->battlerIdTemp].form == (u8) CHERRIM_SUNNY) {
+                ctx->battleMons[ctx->battlerIdTemp].form = (u8) CHERRIM_CLOUDY;
+                *script = 262;
+                ret = TRUE;
+                break;
+            }
+        }
+        if (ctx->battleMons[ctx->battlerIdTemp].species == SPECIES_ARCEUS &&
+            ctx->battleMons[ctx->battlerIdTemp].hp &&
+            GetBattlerAbility(ctx, ctx->battlerIdTemp) == ABILITY_MULTITYPE) {
+            form = GetArceusTypeByHeldItemEffect(GetItemAttr(ctx->battleMons[ctx->battlerIdTemp].item, 1, HEAP_ID_BATTLE));
+            if (ctx->battleMons[ctx->battlerIdTemp].form != form) {
+                ctx->battleMons[ctx->battlerIdTemp].form = form;
+                *script = 262;
+                ret = TRUE;
+                break;
+            }
+        }
+        if (ctx->battleMons[ctx->battlerIdTemp].species == SPECIES_GIRATINA &&
+            ctx->battleMons[ctx->battlerIdTemp].hp &&
+            ctx->battleMons[ctx->battlerIdTemp].form == GIRATINA_ORIGIN) {
+            if ((ctx->battleMons[ctx->battlerIdTemp].status2 & STATUS2_TRANSFORMED) ||
+                (!(BattleSystem_GetBattleFlags(bsys) & 0x80) && ctx->battleMons[ctx->battlerIdTemp].item != ITEM_GRISEOUS_ORB)) {
```

Comment:
constant for battle flags?

Reply:
Dont know what they mean but since battle flag is a set name I'll add placeholder consts